src/MusicalScore/ScoreIO/MusicSheetReader.ts

```diff
@@ -611,8 +611,13 @@ export class MusicSheetReader /*implements IMusicSheetReader*/ {
         }
         let paperHeight: number = 0;
         let topSystemDistance: number = 0;
-        const defi: string = root.element("defaults").element("page-layout").element("page-height").value;
-        paperHeight = parseFloat(defi);
+        try {
+            const defi: string = root.element("defaults").element("page-layout").element("page-height").value;
```

Comment:
Again: check for if element(xyz) === undefined instead of try catch. It gets a longish expression but not as costly as try catch

Reply:
that alternative is checking for undefined 4 times, probably in 4 nested if-checks. we can do that, not sure if it's better or prettier.

Reply:
I would also say: keep it simple now and don't optimize things too early. Let's see if your current changes do the job for us and improve it if we see problems.

Reply:
I'm not even sure the if-checks would perform better. We'd have to repeat each root.element call in each nested if loop (1), or store each in a new variable (2), and then the code really gets ugly.
Also, the try block is only performed once for each score.

method 1:
```javascript
if (root.element("defaults")) {
  if (root.element("defaults").element("page-layout")) {
    // ...
```

method 2:
```javascript
const elementDefaults: IXmlElement = root.element("defaults");
if (elementDefaults !== undefined) {
  const elementPageLayout: IXmlElement = elementDefaults.element("page-layout");
  if (elementPageLayout !== undefined {
    // ...
```

Reply:
In Kotlin, this would be easy: root?.elements("defaults")?... which returns null if any of the calls are null.